### PR TITLE
Ch5&6 ch3 harmonization

### PR DESCRIPTION
**Pull request by kbjarkefur**
Harmonize new chapter 3 with chapter 5 and 6

### CODE REVIEW

**Comment by luizaandrade** [comment]
@luizaandrade check raw data definition

File: chapters/3-measurement.tex [hunk 1]
```diff
@@ -246,6 +246,16 @@ \subsection{Master datasets}
 You should not do anything with that dataset until
 you have successfully merged
 the project IDs from the master dataset.
+When adding new observations to the master datasets,
+always convince yourself beyond reasonable doubt
+that the new observation is indeed a new observation.
+Sometimes you will encounter different spellings in identifiers
+and identifiers such as addresses may become outdated.
+
+Since the master datasets is the authoritative source
```
Comment (bbdaniels) [inline chapters/3-measurement.tex:255]:
```suggestion
Since the master dataset is the authoritative source
```

File: chapters/3-measurement.tex [hunk 2]
```diff
@@ -246,6 +246,16 @@ \subsection{Master datasets}
...
+of the project ID and all research variables,
```
Comment (bbdaniels) [inline chapters/3-measurement.tex:256]:
```suggestion
of the project ID and all research variables
for the corresponding unit of observation,
```

File: chapters/5-processing.tex [hunk 3]
```diff
@@ -100,32 +107,37 @@ \subsection{Unique identifiers}
...
+	original data collected by the research team.
+}
+
+It is also possible for a raw dataset to not include an unique identifier,
+or that the identifier is not a suitable \textbf{project ID}.\sidenote{
+	\textbf{Project ID:} The main ID used in your project to identify
+	observations.
+	You should never have multiple project IDs for the same unit of observation.
+	The project ID must uniquely and fully identify all observations in the project.
+	See \url{https://dimewiki.worldbank.org/ID\_Variable\_Properties} for more details.}
```
Comment (bbdaniels) [inline chapters/5-processing.tex:119]: Why are we repeating the definition here?

File: chapters/3-measurement.tex [hunk 4]
```diff
@@ -127,19 +127,24 @@ \subsection{Data linkage table}
...
+	\textbf{Project ID:} The main ID used in your project to identify
+	observations.
+	You should never have multiple project IDs for the same unit of observation.
+	The project ID must uniquely and fully identify all observations in the project.
+	See \url{https://dimewiki.worldbank.org/ID\_Variable\_Properties} for more details.}
+variable for that unit of observation.
 It is important to include both plain-language terminology
 as well as technical file and variable names here.
-For example, the \texttt{hh_baseline0_listmap.csv} dataset
+For example, the \texttt{hh\_baseline\_listmap.csv} dataset
```
Comment (bbdaniels) [inline chapters/3-measurement.tex:139]:
```suggestion
For example, the \texttt{hh\_baseline2020\_listmap.csv} dataset
```

The `0` was intentional to show extraneous information, is this better?

**Review by bbdaniels — CHANGES_REQUESTED** [review]
Suggestions here!

File: chapters/3-measurement.tex (hunk 4 above)
Comment (kbjarkefur) [reply chapters/3-measurement.tex:139]: ok, this is better. It look like a typo to me

File: chapters/5-processing.tex (hunk 3 above)
Comment (kbjarkefur) [reply chapters/5-processing.tex:119]: As it is the first time it is mentioned in this chapter

Comment (kbjarkefur) [reply chapters/5-processing.tex:119]: Should it maybe be an explanation but not the full definition?

Comment (bbdaniels) [reply chapters/5-processing.tex:119]: I would think we define only once in book and then can point to wiki in subsequent chapters. Otherwise we will create maintenance problems whenever a definition is updated, needing to do it consistently

**Comment by bbdaniels** [comment]
@kbjarkefur Is this ready to merge with those fixes?

File: chapters/5-processing.tex (hunk 3 above)
Comment (mariaruth) [reply chapters/5-processing.tex:119]: Agree with Ben, should only define terms once in the book, but linking to a wiki page for later entries makes sense (or mention here that this is discussed in ch3)

File: chapters/5-processing.tex [hunk 5]
```diff
@@ -100,32 +107,37 @@ \subsection{Unique identifiers}
...
+Suitable project IDs should, for example, not involve long strings
+that are difficult to work with, such as a person's name,
+or be an ID that is known outside the research team.
```
Comment (mariaruth) [inline chapters/5-processing.tex:122]:
```suggestion
or be an ID that is used outside the research team.
```

File: chapters/5-processing.tex [hunk 6]
```diff
@@ -100,32 +107,37 @@ \subsection{Unique identifiers}
...
-Part of the process of understanding the level of observation in a dataset
-and finding or creating its unique identifier is to
-make sure that each observation is represented in the master dataset by a unique ID value.
-If you encounter any observations that are not listed in the master dataset,
-you should assign them a unique ID and add them to the master dataset.
-Always convince yourself beyond reasonable doubt that the new observation
-is indeed a new observation,
-and not one that is already listed, only with a different ID value.
+If a project ID does not exist, 
+then you need to create it,
```
Comment (mariaruth) [inline chapters/5-processing.tex:132]:
```suggestion
then you need to generate one,
```

File: chapters/5-processing.tex [hunk 7]
```diff
@@ -100,32 +107,37 @@ \subsection{Unique identifiers}
...
+add it to the master dataset, 
+and then merge it to the raw data.
```
Comment (mariaruth) [inline chapters/5-processing.tex:134]:
```suggestion
and then merge it back into the raw data.
```

File: chapters/6-analysis.tex [hunk 8]
```diff
@@ -147,13 +147,19 @@ \subsection{Fitting construction into the data workflow}
 in a unified script, separate from the analysis code, helps
 avoid this and ensures consistency across different outputs.
 
-\subsection{Integrating different data sources}
+\subsection{Combining different data sources}
```
Comment (mariaruth) [inline chapters/6-analysis.tex:150]: Why the change? data integration is the technical term, this is not a case of overly-difficult vocabulary

File: chapters/6-analysis.tex [hunk 9]
```diff
@@ -147,13 +147,19 @@ \subsection{Fitting construction into the data workflow}
...
 
 % When merging is necessary and how to start thinking about it
 To create the analysis dataset,
-it is typically necessary to combine or merge information
-from different data sources.\sidenote{
-	\url{https://dimewiki.worldbank.org/Data_Integration}}
+it is typically necessary to combine information
+from different data sources.
```
Comment (mariaruth) [inline chapters/6-analysis.tex:155]: here is probably where we want to link to the data integration page on the dime wiki (currently just a stub)

File: chapters/6-analysis.tex [hunk 10]
```diff
@@ -147,13 +147,19 @@ \subsection{Fitting construction into the data workflow}
...
+As discussed in Chapter 3, 
+this process should be documented and 
+communicated about using \textbf{data flowcharts},\sidenote{
```
Comment (mariaruth) [inline chapters/6-analysis.tex:158]:
```suggestion
this process should be documented using \textbf{data flowcharts},\sidenote{
```

File: chapters/5-processing.tex [hunk 11]
```diff
@@ -100,32 +107,37 @@ \subsection{Unique identifiers}
...
+	The project ID must uniquely and fully identify all observations in the project.
+	See \url{https://dimewiki.worldbank.org/ID\_Variable\_Properties} for more details.}
+Suitable project IDs should, for example, not involve long strings
+that are difficult to work with, such as a person's name,
+or be an ID that is known outside the research team.
+In such cases, cleaning begins by 
+adding a project ID to the raw data.
+If a project ID already exists, 
+for this unit of observation,
+then you should carefully merge it
```
Comment (mariaruth) [inline chapters/5-processing.tex:127]: the sidenote about terminology (merge vs join) currently on line 166 should move up here, as we use the 'term' merge from here on

File: chapters/6-analysis.tex [hunk 12]
```diff
@@ -147,13 +147,19 @@ \subsection{Fitting construction into the data workflow}
...
+	\url{https://dimewiki.worldbank.org/Data\_Flow\_Charts}}
+and different data sources should only be combined
+in accordance with the data linkage table.\sidenote{
+	\url{https://dimewiki.worldbank.org/wiki/Data\_Linkage\_Table}}
 For example, you may merge administrative data with survey data
 in order to include demographic information in your analysis,
 or you may want to integrate geographic information
```
Comment (mariaruth) [inline chapters/6-analysis.tex:165]: the sidenote that starts in the next line should move up (see my earlier comment for a suggestion on where)

File: chapters/6-analysis.tex [hunk 13]
```diff
@@ -171,10 +177,9 @@ \subsection{Integrating different data sources}
...
 % How to think about merges
-Chapter 3 discussed how to merge datasets
-based on their units of observation and identifying variables.
 When writing the code to implement merge operations,
-a few steps can help avoid mistakes.
+a few steps can help avoid mistakes.\sidenote{\url{
+		https://dimewiki.worldbank.org/Data\_Integration}}
 First, before writing code to combine the datasets,
 write pseudocode to understand which observations you expect to be
 matched or not, and why.
```
Comment (mariaruth) [inline chapters/6-analysis.tex:185]: this should also refer back to the data linkage table, as merge outcome expectations should be clear from information there, e.g. which source will have leftover observations, whether merge will be 1-1 or m-1 or 1-m

**Review by mariaruth — CHANGES_REQUESTED** [review]
A few additional revisions suggested in comments

File: chapters/5-processing.tex (hunk 5 above)
Comment (kbjarkefur) [reply chapters/5-processing.tex:122]: I think _known_ is the correct word here. Relating to the discussion we just had on Slack, the key is that no-one knows the ID outside the research team. It is an direct extension of the recommendation that only people listed on the IRB should have access to the identified data. If we want to, we can say that. That "the link between study ID and PII should only be known by people listed on the IRB".

And then I think when we publish the data, for example in a replication package, we can include the study ID, and other researchers can use it to combine data in that replication package in new ways. that usage of the study ID outside the research team is fine, as long as there is no way for them to know how it links to PII.

File: chapters/6-analysis.tex (hunk 8 above)
Comment (kbjarkefur) [reply chapters/6-analysis.tex:150]: I tested a re-structure where integration would come sooner, but it wasnt better, so I changed back but missed that I had committed this. I reverted back to original wording in 653b8453d228ecc48dc78ced1b068797676a3b7b

File: chapters/5-processing.tex (hunk 11 above)
Comment (kbjarkefur) [reply chapters/5-processing.tex]: done in 93480410479541b119828baa56f6dedc28ca3967

File: chapters/6-analysis.tex (hunk 12 above)
Comment (kbjarkefur) [reply chapters/6-analysis.tex:165]: done in 93480410479541b119828baa56f6dedc28ca3967

File: chapters/6-analysis.tex (hunk 9 above)
Comment (kbjarkefur) [reply chapters/6-analysis.tex]: done in 484ae23d18a7f2f5638b1a25d46fb3fcab39cd12

File: chapters/5-processing.tex (hunk 3 above)
Comment (kbjarkefur) [reply chapters/5-processing.tex]: ok, now in the defintion is only in ch 3, but in both ch 3 and 5 we link to the new Project ID wiki page. Done in: 885c47eabf04f59cb9b637ae55d64f5ea9a5fdf9

File: chapters/5-processing.tex (hunk 5 above)
Comment (kbjarkefur) [reply chapters/5-processing.tex:118]: In chapter 3 we already say:

> The project ID must be created by the project team, and the linkage to direct identifiers should only be known to people listed on the IRB.

Comment (kbjarkefur) [reply chapters/5-processing.tex:118]: @mariaruth and @bbdaniels , I updated Maria's original suggestion. Let me know what you think about this.

Comment (bbdaniels) [reply chapters/5-processing.tex:118]: Yes, I like this, and I would change "person's name" on 121 to just "name", since "person's name" is _obviously_ identifying and should almost never be used at all

File: chapters/6-analysis.tex (hunk 10 above)
Comment (kbjarkefur) [reply chapters/6-analysis.tex]: done in 74e1ac0a2fe9f1d90ee32d95431aefe588060c63

File: chapters/5-processing.tex (hunk 5 above)
Comment (kbjarkefur) [reply chapters/5-processing.tex:118]: > Yes, I like this, and I would change "person's name" on 121 to just "name", since "person's name" is _obviously_ identifying and should almost never be used at all

done in 9ee3119c5ff4ed94f2b383a1ef6201418736cbb3

File: chapters/6-analysis.tex (hunk 13 above)
Comment (kbjarkefur) [reply chapters/6-analysis.tex:182]: done in 7e4080b820ac09645634ab0a33125a5ed9bdec9c

**Comment by kbjarkefur** [comment]
Thanks for comments! done with edits, please review and resolve and merge when everything is resolved!

**Review by bbdaniels — APPROVED** [review]

**Review by mariaruth — APPROVED** [review]

**Comment by mariaruth** [comment]
I think this is ready to merge, but not sure if @luizaandrade wanted to review first?

**Review by luizaandrade — APPROVED** [review]
